metadata-service/auth-impl/src/main/java/com/datahub/authorization/DataHubAuthorizer.java

```diff
@@ -102,14 +104,17 @@ public AuthorizationResult authorize(@Nonnull final AuthorizationRequest request
     return new AuthorizationResult(request, AuthorizationResult.Type.DENY,  null);
   }
 
-  public List<String> getGrantedPrivileges(final String actorUrn, final Optional<ResourceSpec> resourceSpec) {
+  public List<String> getGrantedPrivileges(final String actor, final Optional<ResourceSpec> resourceSpec) {
 
     // 1. Fetch all policies
     final List<DataHubPolicyInfo> policiesToEvaluate = _policyCache.getOrDefault(ALL, new ArrayList<>());
 
+    Urn actorUrn = UrnUtils.getUrn(actor);
+    final ResolvedResourceSpec actorResolvedResourceSpec = _resourceSpecResolver.resolve(new ResourceSpec(actorUrn.getEntityType(), actor));
```

Comment:
I see the idea here.. nice! 

Do we _need_ to resolve the resource spec at this point, or can we do it later on once we absolutely _know_ that group membership is required. I want to make sure that we avoid making any additional lookups until we absolutely have to

Reply:
So there are two options:
- As it is implemented now, resolving the actor and pass it to the PolicyEngine. It is true that at this point we do not know if group membership is required. However, according to [this line](https://github.com/datahub-project/datahub/blob/master/metadata-auth/auth-api/src/main/java/com/datahub/authorization/FieldResolver.java#L16), no field is resolved until it is explicitly called/obtained.
- The other option implies passing the `_resourceSpecResolver ` to the PolicyEngine class. The PolicyEngine class is initialized in the DatahubAuthorizer constructor while the `_resourceSpecResolver `  in the init function. So if we want to follow this approach, the PolicyEngine class should be initialized in the init method after the resolver. Not sure if this has any implications.